kanidm_rlm_python/kanidmradius/__init__.py

```diff
@@ -0,0 +1,199 @@
+""" kanidm RADIUS module """
+
+import asyncio
+from functools import reduce
+import json
+import logging
+import os
+from pathlib import Path
+import sys
+from typing import Any, Dict, Optional, Union
+
+import aiohttp
+
+from kanidm import KanidmClient
+from kanidm.types import AuthStepPasswordResponse
+from kanidm.utils import load_config
+from kanidm.exceptions import NoMatchingEntries
+
+from . import radiusd
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    stream=sys.stderr,
+    )
+
+# the list of places to try
+config_paths = [
+    os.getenv("KANIDM_RLM_CONFIG", "/data/kanidm"), # container goodness
+    "~/.config/kanidm", # for a user
+    "/etc/kanidm/kanidm", # system-wide
+]
+
+CONFIG_PATH = None
+for config_file_path in config_paths:
+    CONFIG_PATH = Path(config_file_path).expanduser().resolve()
+    if CONFIG_PATH.exists():
+        break
+
+if (CONFIG_PATH is None) or (not CONFIG_PATH.exists()):
+    logging.error("Failed to find configuration file, checked (%s), quitting!", config_paths)
+    sys.exit(1)
+config = load_config(str(CONFIG_PATH))
+
+COOKIE_JAR = aiohttp.CookieJar()
+KANIDM_CLIENT = KanidmClient(config_file=CONFIG_PATH)
+
+def authenticate(
+    acct: str,
```

Comment:
Holy shit, types!

Reply:
TYPE ALL THE THINGS! (also the `py.typed` file says "I am typed!")